src/gemini_supply/orchestrator.py

```diff
@@ -174,48 +172,50 @@ def load_init_scripts():
   ]
 
 
-async def _denature_search_results_page(page: playwright.async_api.Page) -> None:
-  url = page.url
-  parsed = urlparse(url)
-
-  termcolor.cprint(f"[denature] {url}", color="light_grey")
-  if parsed.path != "/en/online-grocery/search":
-    return
-
-  termcolor.cprint("[denature] On search results page", color="cyan")
-
-  # Check for and click any visible overlay close buttons
-  close_buttons = page.locator("a.close-overlay-box")
-  count = await close_buttons.count()
-  termcolor.cprint(f"[denature] Found {count} overlay close button(s), clicking...", color="cyan")
-  for i in range(count):
-    button = close_buttons.nth(i)
-    if await button.is_visible():
-      await button.click()
-      termcolor.cprint(f"[denature] Clicked overlay close button {i + 1}/{count}", color="green")
-
-  # Replace all a.product-details-link with span elements
-  product_links = page.locator("a.product-details-link")
-  link_count = await product_links.count()
-  if link_count > 0:
-    termcolor.cprint(
-      f"[denature] Replacing {link_count} product link(s) with spans...", color="cyan"
-    )
-    await page.evaluate("""(() => {
-      const links = document.querySelectorAll('a.product-details-link');
-      links.forEach(link => {
-        const span = document.createElement('span');
-        span.className = link.className;
-        span.innerHTML = link.innerHTML;
-        Array.from(link.attributes).forEach(attr => {
-          if (attr.name !== 'href') {
-            span.setAttribute(attr.name, attr.value);
-          }
+def _make_denature_delegate(log: ActivityLog):
```

Comment:
Hmmm...can we use functools.partial instead of this? Feels more direct, and I don't like seeing these changeless diffs.